horovod/mxnet/__init__.py

```diff
@@ -167,17 +182,22 @@ def _allreduce_grads(self):
 
                 for entries in entries_by_dtype.values():
                     grads, names = zip(*entries)
-                    grouped_allreduce_(tensors=grads, average=False, name="{}:{}".format(names[0], names[-1]), priority=-i,
+                    tensors_compressed, ctxs = zip(*[self._compression.compress(grad) for grad in grads])
+                    grouped_allreduce_(tensors=tensors_compressed, average=False, name="{}:{}".format(names[0], names[-1]), priority=-i,
                                        prescale_factor=1.0 / self._gradient_predivide_factor)
+                    grads = [self._compression.decompress(t, ctx) for t, ctx in zip(tensors_compressed, ctxs)]
```

Comment:
Same issue here. Pretty sure this is just updating a local list but not actually modifying the real parameter gradients.